icu4c/source/common/unicode/uvernum.h

```diff
@@ -111,7 +111,7 @@
 
 #ifndef U_ICU_ENTRY_POINT_RENAME
 #ifdef U_HAVE_LIB_SUFFIX
-#   if !U_DISABLE_VERSION_SUFFIX
+#   ifndef U_DISABLE_VERSION_SUFFIX
```

Comment:
This does not look right: It changes from "do this if macro value is 0" to "do this if the macro is not defined".
I think it should be
```
#if !defined(macro) || !macro
```

Reply:
yes, you are right.

Reply:
Right, thanks for noticing. Yes, it's a change in how these macros are handled, but I think this is okay because:

1. This macro is brand new in ICU 64
2. The rest of the file seems to use ifdef instead of if

I also changed some debug macros from if directly to ifdef. For macros that were older and didn't look like debugging, I used the defined && true syntax like you suggested.

Let me know if that sounds reasonable.

Reply:
it might make more sense for documentation purposes to add this above this block:

```c
#ifndef U_DISABLE_VERSION_SUFFIX
#define U_DISABLE_VERSION_SUFFIX 0
#endif 
```

then keep the rest of the code as it was

Reply:
I like Steven's suggestion.

I noticed switching the logic for the debug macros, and I don't mind it for those.

Reply:
Amended as requested.